test/load-generator/boulder-calls.go

```diff
@@ -389,7 +379,7 @@ func completeAuthorization(authz *core.Authorization, s *State, ctx *context) er
 // correct authorization state an error is returned. If no error is returned
 // then the authorization is valid and ready.
 func pollAuthorization(authz *core.Authorization, s *State, ctx *context) error {
-	authzURL := fmt.Sprintf("%s/acme/authz/%s", s.apiBase, authz.ID)
+	authzURL := authz.ID
```

Comment:
Huh, this seems like it shouldn't work? How does `getAuthorization` get the rest of the URL?

Reply:
You're right. I think this is another place where I had to do a lot of refactoring to make something sensible and when I tried to pull out just the directory change I missed this. I'll address.

Reply:
I was wrong about you being right. It does work it's just not straightforward because I jammed it into the existing code. That makes sense: I did actually test that this branch issues certificates against Boulder and it wouldn't work if the polling were broken.

The `fulfillOrder` action takes a pending order from the `context`. It calls `getAuthorization(State, url)` for each fully qualified authz `url` from the pending order. At the end of `getAuthorization` the `url` that it was provided (from the order) is assigned to the returned `core.Authorization.ID` field.

The line you flagged is in `pollAuthorization`, which is called by `completeAuthorization` after the call to `getAuthorization`. By the time `authzURL := authz.ID` is run the `ID` field is the authorization's URL from the pending order.

Reply:
Ah, that makes sense. I totally missed that `getAuthorization` sets the ID to the URL. It feels a bit... wrong, but I can't think of a better solution without making a bunch of signature changes, so whatever.

Thanks for walking me through it!